### ISSUE TITLE
Low contrast on primary action hover when using outline prop

### ISSUE THREAD

**Issue by laurenmayers** [Shopify/polaris#4054]
## Issue summary
A `primaryAction` button with `outline` has very low contrast on hover.  

## Expected behavior
The button should have dark text so that the text is visible and it passes contrast standards. 

## Actual behavior
Button in hovered state:
<img width="574" alt="Screen Shot 2021-03-10 at 2 12 47 PM" src="https://user-images.githubusercontent.com/35752699/110684085-b5d52400-81aa-11eb-9eca-d59cd4ff3465.png">

## Steps to reproduce the problem
Add an `outline: true` prop to a `primaryAction` on a Polaris component and observe on hover

## Reduced test case
Sandbox: https://codesandbox.io/s/gifted-keller-c05mg?file=/App.js

## Specifications

- Are you using the React components? (Y/N): Y

```bash
 System:
    OS: macOS 11.2.2
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 925.33 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.20.1 - ~/.nvm/versions/node/v12.20.1/bin/node
    Yarn: 1.22.5 - ~/.dev/yarn/1.22.5/bin/yarn
    npm: 6.14.10 - ~/.nvm/versions/node/v12.20.1/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 89.0.4389.82
    Firefox Developer Edition: 84.0
    Safari: 14.0.3
  npmPackages:
    @shopify/polaris: 6.0.1 => 6.0.1
    react: ~16.14.0 => 16.14.0
    react-dom: ~16.14.0 => 16.14.0
```
